### ISSUE TITLE
bin symlink goes to .yalc instead of node_modules copy

### ISSUE THREAD

**Issue by jasonkuhrt** [wclr/yalc#156]
I'm working on a package that has a generated aspect.

It has a CLI to run the generation.

When generation runs, the package expects to find itself within node_modules.

The problem is that the `node_modules/.bin` file symlinks (it seems) to the `.yalc` location, rather than the `node_modules/...` location.

In turn this breaks my current case since the generated library parts are going into the wrong directory now:

![image](https://user-images.githubusercontent.com/284476/109209696-d7311b80-7779-11eb-94f4-9fefce954442.png)

In the above `.yalc` should be `node_modules`.

**Comment by wclr** [comment]
Sorry, I'm not following. Try to explain the steps you make to get the issue, or maybe there could be a repro example.

**Comment by jasonkuhrt** [comment]
1. Create a package `A` with a bin specified in package.json
2. `yalc add A` in some other project `B`
3. `yalc push` `A`
4. `A` gets put into `B` like normal, but its `bin` script is symlinked into `B/.yalc/A/`... it should be pointing toward `B/node_modules/A`.

**Comment by wclr** [comment]
Right. There seems [not need bin sysmlinks are created](https://github.com/wclr/yalc/blob/543d26de61b85a03e098a2adc0e7de448306e83c/src/add.ts#L165) while `add` operation. Will fix it soon.

**Comment by jasonkuhrt** [comment]
Also @wclr I am seeing issues with regard to needing to run `chmod +x` over the bin file. Maybe related to https://github.com/wclr/yalc/issues/21. I assume package managers do this `chmod +x` automatically and its just that `yalc` is not right now?

**Comment by wclr** [comment]
> I am seeing issues with regard to needing to run chmod +x over the bin file.

What bin file? Where it is located?


Actually, I'm a little bit confused with this thing with .bin scripts, which was added but a contributor's PR. I generally never used and run into it, cause I use `--pure` adding. I believe linking of bin script from .yalc folder need only if `yalc link` or `yalc add --link` (yarn's `link:` protocol), if just `yalc add` is used then PM should take care of installing bin scripts.

**Comment by jasonkuhrt** [comment]
I used `yalc push --force` and `yalc add`.

> What bin file? Where it is located?

In the package where  run `yalc push --force`. The script is listed in `package.json` `bin` field.

**Comment by wclr** [comment]
> Location should not matter. Whatever the value of bin in package.json is.

I mean is it related to [this code](https://github.com/wclr/yalc/blob/543d26de61b85a03e098a2adc0e7de448306e83c/src/add.ts#L169)?

Do you have the error stack?

**Comment by jasonkuhrt** [comment]
> I mean is it related to this code?

Yeah think so, but not sure.
